assayist/processor/base.py

```diff
@@ -0,0 +1,184 @@
+            'checksum_source': content.Checksum.CHECKSUM_SOURCES['unsigned']})[0]
+
+        artifact.checksums.connect(checksum_node)
+        return artifact
+
+    def get_or_create_rpm_artifact_from_rpm_info(self, rpm_info):
+        """
+        Create or update an Artifact for this rpm from a dictionary.
+
+        :param str rpm_info: A dictionary of information, like one that comes from brew.
+                             Must contain the fields used in get_or_create_rpm_artifact.
+        :return: an Artifact object
+        :rtype: assayist.common.models.content.Artifact
+        """
+        return self.get_or_create_rpm_artifact(
+            id=rpm_info['id'],
+            name=rpm_info['name'],
+            version=rpm_info['version'],
+            release=rpm_info['release'],
+            arch=rpm_info['arch'],
+            checksum=rpm_info['payloadhash'])
+
+    def get_or_create_archive_artifact(self, archive_id, filename, arch, type, checksum):
+        """
+        Create or update an Artifact for this archive.
+
+        :param str archive_id: the archives's id
+        :param str filename: the archive's filename, eg 'maven.jar'
+        :param str arch: the archive's architecture, eg 'x86_64' or None
+        :param str type: the archive's type, eg 'maven'
+        :param str checksum: the archive's checksum, eg 'deadbeef1234'
+        :return: an Artifact object
+        :rtype: assayist.common.models.content.Artifact
+        """
+        if type == 'image':
+            type = content.Artifact.TYPES['container']
+        elif type == 'maven':
+            type = content.Artifact.TYPES['maven']
+        else:
+            type = content.Artifact.TYPES['other']
+
+        artifact = content.Artifact.create_or_update({
+            'archive_id': archive_id,
+            'type_': type,
+            'architecture': arch,
+            'filename': filename})[0]
+
+        checksum_node = content.Checksum.create_or_update({
+            'checksum': checksum,
+            'algorithm': content.Checksum.guess_type(checksum),
+            'checksum_source': content.Checksum.CHECKSUM_SOURCES['unsigned']})[0]
+
+        artifact.checksums.connect(checksum_node)
+        return artifact
+
+    def get_or_create_source_location(self, url, canonical_version):
+        """
+        Create or update SourceLocation.
+
+        :param str url: the url and possibly commit hash of the source location
+        :param str canonical_version: (optional) the version of the componenet that
+                                      corresponds with this commit id
+        :return: a SourceLocation object
+        :rtype: assayist.common.models.source.SourceLocation
+        """
+        sl = source.SourceLocation.get_or_create({'url': url})[0]
+        if canonical_version:
+            sl.canonical_version = canonical_version
+        return sl
+
+    def get_or_create_component(self, canonical_namespace, canonical_name, canonical_type):
```

Comment:
This method doesn't provide any value. Could we please just replace calls to this method with direct calls to the library?